src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,241 @@
+""" show_dampening.py
+    supports commands:
+        * show im dampening
+        * show im dampening {interface}
+"""
+
+# Python
+import re
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Optional
+
+# parser utils
+from genie.libs.parser.utils.common import Common
+
+# =============================================
+# Parser for 'show im dampening'
+# =============================================
+
+class ShowImDampeningSchema(MetaParser):
+    schema = {
+                'interface': {
+                    Any(): {
+                    'protocol': {
+                        Any(): {
+                        'capsulation': str,
+                        'penalty': int,
+                        'suppressed': str,
+                        }
+                    }
+                }
+            }
+        }
+                    
+class ShowImDampening(ShowImDampeningSchema):
+    """ Parser for show im dampening"""
+    
+    cli_command = 'show im dampening'
+
+    def cli(self,output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        result_dict.setdefault('interface', {})
+        
+        # GigabitEthernet0/0/0/1                                               2389 YES
+        # POS0/2/0/0                  <base>             ppp                      0 NO 
+        p1 = re.compile(r'^(?P<interface>.{1,26}) +(?P<prot>.{1,20}) +(?P<cap>.{1,10}) +(?P<pen>\d+) +(?P<sup>YES|NO)')
```

Comment:
I feel this regex may be a little risky (unless you are confident about the spacing) and it seems to catch a lot of whitespace:
![image](https://user-images.githubusercontent.com/1134387/96029776-1db85600-0e29-11eb-9dc3-a1bd08cca6b3.png)

`^(?P<interface>\S+) +(?P<prot>\S+)? +(?P<cap>\S+)? +(?P<pen>\d+) +(?P<sup>YES|NO)` Might be better:
![image](https://user-images.githubusercontent.com/1134387/96029991-6bcd5980-0e29-11eb-818d-5bbf79cdcc1a.png)

Reply:
Nice one Daniel, that suggestion is much better. I didn't know about the ? as an optional field. Thank you, I'll use that going forward.